src/main/java/com/airlines/dao/CrewMemberDaoImpl.java

```diff
@@ -0,0 +1,145 @@
+        insertStatement.setString(3, String.valueOf(crewMember.getPosition()));
+        insertStatement.setDate(4, Date.valueOf(crewMember.getBirthday()));
+        insertStatement.setString(5, String.valueOf(crewMember.getCitizenship()));
+        return insertStatement;
+    }
+
+    private void executeUpdate(PreparedStatement insertStatement, String errorMessage) throws SQLException {
+        int rowsAffected = insertStatement.executeUpdate();
+        if (rowsAffected == 0) {
+            throw new DaoOperationException(errorMessage);
+        }
+    }
+
+    private int fetchGeneratedId(PreparedStatement insertStatement) throws SQLException {
+        ResultSet generatedKeys = insertStatement.getGeneratedKeys();
+
+        if (generatedKeys.next()) {
+            return generatedKeys.getInt(1);
+        } else {
+            throw new DaoOperationException("Can not obtain an account ID");
+        }
+    }
+
+    public CrewMember findOne(int id) {
+        try (Connection connection = dataSource.getConnection()) {
+            return findCrewMemberById(id, connection);
+        } catch (SQLException e) {
+            throw new DaoOperationException(String.format("Cannot find Account by id = %d", id), e);
+        }
+    }
+
+    private CrewMember findCrewMemberById(int id, Connection connection) throws SQLException {
+        PreparedStatement selectByIdStatement = prepareSelectByIdStatement(id, connection);
+        ResultSet resultSet = selectByIdStatement.executeQuery();
+        resultSet.next();
+        return parseRow(resultSet);
+    }
+
+    private PreparedStatement prepareSelectByIdStatement(int id, Connection connection) {
+        try {
+            PreparedStatement selectByIdStatement = connection.prepareStatement(SELECT_CREWMEMBER_BY_ID_SQL);
+            selectByIdStatement.setLong(1, id);
+            return selectByIdStatement;
+        } catch (SQLException e) {
+            throw new DaoOperationException("Cannot prepare statement to select account by id", e);
+        }
+    }
+
+    private CrewMember parseRow(ResultSet resultSet) throws SQLException {
+        CrewMember crewMember = CrewMember.builder()
+                .withId(resultSet.getInt("id"))
+                .withFirsName(resultSet.getString("first_name"))
+                .withLastName(resultSet.getString("last_name"))
+                .withBirthday(resultSet.getDate("birthday").toLocalDate())
+                .withPosition(Position.BOARD_CONDUCTOR)
+                .withPosition(Position.FIRST_PILOT)
+                .withPosition(Position.STEWARDESS)
+                .withPosition(Position.FIRST_PILOT)
+                .withCitizenship(Citizenship.AUSTRIA)
+                .withCitizenship(Citizenship.AUSTRALIA)
+                .withCitizenship(Citizenship.GERMANY)
+                .withCitizenship(Citizenship.NEW_ZEALAND)
+                .withCitizenship(Citizenship.UK)
+                .withCitizenship(Citizenship.USA)
```

Comment:
Imagine you have Cat class:

```
@Builder
class Cat {
   private int age;
   private String name;
}
```

this method is wehere you get information (row / resultset) from database and parse into your entity.
so you have Cat("murka", 25) from database: and doing next:

Cat.builder()
   .name("tuzik")
   .name("murka")
   .name("furryik")
    .age(10)
    .age(5)
    .age(10)
    .build();
    
    but all you need just:
     Cat.build()
        .name(resultSet.getString("name"))
        .age(resultSet.getString("age))
        .build();
        
        Enum is just ordinary string (varchar) in database. All you need here - is just get string from database and parse it into enum:
         .withPosition(Position.valueOf(resultSet.getString("position"))
          smth like that